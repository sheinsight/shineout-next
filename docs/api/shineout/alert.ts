const api = JSON.parse('[{"title":"Alert","properties":[{"name":"className","tag":{"cn":"自定义类名","en":"Custom class name","default":"","version":""},"required":false,"type":"string "},{"name":"style","tag":{"cn":"自定义样式","en":"Custom style","default":"","version":""},"required":false,"type":"CSSProperties "}],"cn":"","en":""}]');
export default api;
