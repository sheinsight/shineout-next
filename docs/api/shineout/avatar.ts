const api = JSON.parse('[{"title":"Avatar","properties":[{"name":"className","tag":{"cn":"自定义类名","en":"Custom class name","default":"","version":""},"required":false,"type":"string "},{"name":"style","tag":{"cn":"自定义样式","en":"Custom style","default":"","version":""},"required":false,"type":"CSSProperties "},{"name":"size","tag":{"cn":"不同尺寸","en":"There are three built-in size: small、default、large.","default":"\\\"default\\\"","version":""},"required":false,"type":"\\\"small\\\" | \\\"large\\\" | \\\"default\\\""},{"name":"alt","tag":{"cn":"图像无法显示时的替代文本","en":"The alternative text for the image","default":"","version":""},"required":false,"type":"string "},{"name":"src","tag":{"cn":"图片类头像的资源地址或者图片元素","en":"The address of the image or the image element of the image class avatar","default":"","version":""},"required":false,"type":"ReactNode"},{"name":"icon","tag":{"cn":"设置头像的自定义图标","en":"Set the custom icon of the avatar","default":"","version":""},"required":false,"type":"ReactNode"},{"name":"children","tag":{"cn":"指定头像的形状","en":"Specifies the shape of the avatar","default":"\\\"circle\\\"","version":""},"required":false,"type":"ReactNode"},{"name":"shape","tag":{"cn":"头像文本内容，会根据内容尺寸自适应计算缩放比例","en":"The text content of the avatar, the scaling ratio will be calculated according to the content size","default":"","version":""},"required":false,"type":"\\\"circle\\\" | \\\"square\\\" "},{"name":"srcSet","tag":{"cn":"设置图片类头像响应式资源地址","en":"Set the responsive resource address of the image class avatar","default":"","version":""},"required":false,"type":"string "},{"name":"gap","tag":{"cn":"字符类型距离左右两侧边界单位像素","en":"The distance between the left and right edges of the character type in pixels","default":"","version":""},"required":false,"type":"number "},{"name":"draggable","tag":{"cn":"图片是否允许拖动","en":"Whether the image is allowed to be dragged","default":"true","version":""},"required":false,"type":"boolean "},{"name":"crossOrigin","tag":{"cn":"CORS 属性设置","en":"CORS attribute setting","default":"","version":""},"required":false,"type":"\\\"anonymous\\\" | \\\"use-credentials\\\" "},{"name":"onError","tag":{"cn":"图片加载失败的事件，返回 false 会关闭组件默认的 fallback 行为","en":"The event that the image fails to load, returning false will close the component\\\"s default fallback behavior","default":"","version":""},"required":false,"type":"(() => boolean) "}],"cn":"","en":"","sort":"0"},{"title":"Avatar.Group","properties":[{"name":"className","tag":{"cn":"自定义类名","en":"Custom class name","default":"","version":""},"required":false,"type":"string "},{"name":"style","tag":{"cn":"自定义样式","en":"Custom style","default":"","version":""},"required":false,"type":"CSSProperties "},{"name":"size","tag":{"cn":"不同尺寸","en":"There are three built-in size: small、default、large.","default":"\\\"default\\\"","version":""},"required":false,"type":"\\\"small\\\" | \\\"large\\\" | \\\"default\\\""},{"name":"shape","tag":{"cn":"头像文本内容，会根据内容尺寸自适应计算缩放比例","en":"The text content of the avatar, the scaling ratio will be calculated according to the content size","default":"","version":""},"required":false,"type":"\\\"circle\\\" | \\\"square\\\" "},{"name":"max","tag":{"cn":"设置头像组展示头像的最大数量","en":"Set the maximum number of avatars displayed in the avatar group","default":"","version":""},"required":false,"type":"number "},{"name":"renderMax","tag":{"cn":"自定义渲染隐藏的头像节点，hideAvatarNodes 为隐藏的节点 dom 数组，max 为隐藏的节点数量","en":"Custom rendering of hidden avatar nodes, hideAvatarNodes is an array of hidden node dom, max is the number of hidden nodes","default":"","version":""},"required":false,"type":"((hideAvatarNodes: ReactNode[], max: number) => ReactNode) "},{"name":"children","tag":{"cn":"头像节点","en":"Avatar node","default":"","version":""},"required":true,"type":"ReactNode"},{"name":"popover","tag":{"cn":"用于展示隐藏头像的 Popover 相关配置，配置属性为 Popover 部分属性: position、zIndex、getPopupContainer、adjust","en":"The event that the image fails to load, returning false will close the component\\\"s default fallback behavior","default":"","version":""},"required":false,"type":"Pick<PopoverProps, \\\"position\\\" | \\\"zIndex\\\" | \\\"getPopupContainer\\\" | \\\"adjust\\\"> "}],"cn":"","en":"","sort":"0"}]');
export default api;
