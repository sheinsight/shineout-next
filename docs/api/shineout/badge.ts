const api = JSON.parse('[{"title":"Badge","properties":[{"name":"className","tag":{"cn":"自定义类名","en":"Custom class name","default":"","version":""},"required":false,"type":"string "},{"name":"style","tag":{"cn":"自定义样式","en":"Custom style","default":"","version":""},"required":false,"type":"CSSProperties "},{"name":"children","tag":{"cn":"内容","en":"Content","default":"","version":""},"required":false,"type":"ReactNode"},{"name":"color","tag":{"cn":"自定义徽标颜色","en":"Custom badge color","default":"","version":""},"required":false,"type":"string "},{"name":"count","tag":{"cn":"展示的数字，大于 overflowCount 时显示 ${overflowCount}+，为 0 时隐藏","en":"Number to show, show ${overflowCount}+ when it is greater than overflowCount, hide when it is 0","default":"","version":""},"required":false,"type":"ReactNode"},{"name":"dot","tag":{"cn":"小点模式，开启后不展示数字","en":"Dot mode","default":"false","version":""},"required":false,"type":"boolean "},{"name":"offset","tag":{"cn":"偏移量","en":"Offset of the badge","default":"","version":""},"required":false,"type":"[number, number] "},{"name":"overflowCount","tag":{"cn":"封顶数值","en":"Offset of the badge","default":"","version":""},"required":false,"type":"number "},{"name":"showZero","tag":{"cn":"当数值为 0 时，是否展示 Badge","en":"Whether to show Badge when the number is 0","default":"","version":""},"required":false,"type":"boolean "},{"name":"status","tag":{"cn":"徽标状态","en":"Whether to show Badge when the number is 0","default":"","version":""},"required":false,"type":"\\\"success\\\" | \\\"processing\\\" | \\\"default\\\" | \\\"error\\\" | \\\"warning\\\" "},{"name":"text","tag":{"cn":"状态点文本，仅在配置了 status 属性后生效","en":"Status point text, only valid after configuring the status property","default":"","version":""},"required":false,"type":"ReactNode"},{"name":"size","tag":{"cn":"尺寸，仅在非 dot 模式下生效，支持 small 和 default 两种尺寸","en":"Size, only valid in non-dot mode, supports two sizes: small and default","default":"","version":""},"required":false,"type":"\\\"default\\\" | \\\"small\\\" "}],"cn":"","en":"","sort":"0"}]');
export default api;
