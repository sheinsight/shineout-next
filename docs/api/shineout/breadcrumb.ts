const api = JSON.parse('[{"title":"Breadcrumb","properties":[{"name":"className","tag":{"cn":"自定义类名","en":"Custom class name","default":"","version":""},"required":false,"type":"string "},{"name":"style","tag":{"cn":"自定义样式","en":"Custom style","default":"","version":""},"required":false,"type":"CSSProperties "},{"name":"data","tag":{"cn":"面包屑对象数组,见 data","en":"The array of breadcrumb objects, see data","default":"[]","version":""},"required":false,"type":"any[]"},{"name":"separator","tag":{"cn":"面包屑分隔符,可以是字符串或自定义的元素","en":"A breadcrumb separator which can be strings or custom elements","default":"\\\"/\\\"","version":""},"required":false,"type":"ReactNode"},{"name":"keygen","tag":{"cn":"生成每一项key的辅助方法。为 true 时，以数据项本身作为key，相当于 (d => d)；为函数时，使用此函数返回值；为string时，使用这个string对应的数据值。如 \\\"id\\\"，相当于 (d => d.id)","en":"Key generator.When it is true, the data itself is used as the key equivalent to (d => d);When it is a function, use its return value;When it is a string，ues the value of the string.For example, \\\"id\\\" is the same thing as (d) => d.id.","default":"","version":""},"required":false,"type":"true | ObjectKey<Item> | (data: Item, index?: number ) => string | number"},{"name":"renderItem","tag":{"cn":"自定义渲染","en":"Custom render","default":"","version":""},"required":false,"type":"((value: Item) => ReactNode) "},{"name":"max","tag":{"cn":"最大显示个数","en":"Maximum number of displays","default":"","version":""},"required":false,"type":"number "},{"name":"maxHeight","tag":{"cn":"下拉框最大高度，超出可以内滚","en":"The maximum height of the drop-down box. It can be scrolled if it exceeds the height.","default":"50vh","version":""},"required":false,"type":"string | number "}],"cn":"","en":"","sort":"0"},{"title":"BreadcrumbData","properties":[{"name":"onClick","tag":{"cn":"点击事件","en":"The click event","default":"","version":""},"required":false,"type":"MouseEventHandler<HTMLAnchorElement> "},{"name":"title","tag":{"cn":"显示内容","en":"Displayed content","default":"","version":""},"required":false,"type":"ReactNode"},{"name":"url","tag":{"cn":"链接地址，onClick 属性二选一","en":"Link address","default":"","version":""},"required":false,"type":"string "},{"name":"icon","tag":{"cn":"自定义图标","en":"Custom icon","default":"","version":""},"required":false,"type":"ReactNode"}],"cn":"","en":"","sort":"0"}]');
export default api;
