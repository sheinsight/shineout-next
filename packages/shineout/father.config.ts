import { defineConfig } from 'father';
export default defineConfig({
  extends: '../../.fatherrc.ts',
});
