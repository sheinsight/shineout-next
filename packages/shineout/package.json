{
  "name": "shineout",
  "version": "3.0.0",
  "description": "A components library for React",
  "keywords": [],
  "license": "ISC",
  "author": "",
  "sideEffects": false,
  "main": "./dist/cjs/index.js",
  "module": "./dist/esm/index.js",
  "typings": "./dist/cjs/index.d.ts",
  "scripts": {
    "build": "father build",
    "build:watch": "father dev",
    "test": "echo \"Error: no test specified\" && exit 1"
  },
  "dependencies": {
    "@sheinx/base": "workspace:*",
    "@sheinx/hooks": "workspace:*",
    "@sheinx/shineout-style": "workspace:*"
  },
  "peerDependencies": {
    "classnames": ">=2",
    "immer": ">=10",
    "react": ">=16.8"
  },
  "devDependencies": {
    "jszip": "^3.10.1"
  }
}
