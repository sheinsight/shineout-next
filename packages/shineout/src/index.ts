export { default as Alert } from './alert';
export { default as Button } from './button';
export { default as Checkbox } from './checkbox';
export { default as DatePicker } from './date-picker';
export { default as Dropdown } from './dropdown';
export { default as EditableArea } from './editable-area';
export { default as Form } from './form';
export { default as Icon } from './icon';
export { default as Image } from './image';
export { default as Input } from './input';
export { default as Popover } from './popover';
export { default as Radio } from './radio';
export { default as Rule } from './rule';
export { default as Spin } from './spin';
export { default as Switch } from './switch';
export { default as Tag } from './tag';
export { default as Tabs } from './tabs';
export { default as Textarea } from './textarea';
export { default as Tooltip } from './tooltip';
export { setConfig } from '@sheinx/base';

export * as TYPE from './type';
