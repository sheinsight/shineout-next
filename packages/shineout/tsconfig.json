{
  "extends": "../../tsconfig.json",
  "compilerOptions": {
    "declaration": true,
    "baseUrl": "./",
    "paths": {
      "shineout": ["./src/index"],
      "@sheinx/base": ["../base/src/index"],
      "@sheinx/hooks": ["../hooks/src/index"]
    }
  },
  "include": ["src"]
}
