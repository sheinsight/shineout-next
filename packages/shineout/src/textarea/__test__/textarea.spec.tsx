import { render, fireEvent, cleanup, screen } from '@testing-library/react';
import React from 'react';
import Textarea from '..';
import { Form } from 'shineout';
import mountTest from '../../tests/mountTest';
import {
  attributesTest,
  classTest,
  hasAttributesTest,
  snapshotTest,
  styleTest,
  textContentTest,
  displayTest,
} from '../../tests/utils';
import { classLengthTest } from '../../tests/structureTest';
import TextareaBase from '../__example__/01-base';
import TextareaSize from '../__example__/02-size';
import TextareaAutosize from '../__example__/03-autosize';
import TextareaInfo from '../__example__/04-info';
import TextareaInfoFunction from '../__example__/05-custom';
import TextareaFooter from '../__example__/06-footer';

const SO_PREFIX = 'textarea';
const textareaClassName = `.${SO_PREFIX}-wrapper-0-2-1`;
const textareaFocusClassName = `${SO_PREFIX}-wrapperFocus-0-2-5`;
const textareaSmallClassName = `${SO_PREFIX}-wrapperSmall-0-2-3`;
const textareaShadowClassName = `${SO_PREFIX}-shadow-0-2-15`;
const textareaInfoClassName = `.${SO_PREFIX}-info-0-2-13`;
const textareaInfoErrorClassName = `${SO_PREFIX}-infoError-0-2-14`;
const textareaFooterClassName = `.${SO_PREFIX}-footer-0-2-16`;
const textareaUnderlineClassName = `${SO_PREFIX}-wrapperUnderline-0-2-8`;
const textareaNoBorderClassName = `${SO_PREFIX}-wrapperNoBorder-0-2-9`;

afterEach(cleanup);
describe('Textarea[Base]', () => {
  mountTest(Textarea);
  displayTest(Textarea, 'ShineoutTextarea');
  snapshotTest(<TextareaBase />);
  snapshotTest(<TextareaSize />, 'about size');
  snapshotTest(<TextareaAutosize />, 'about autosize');
  snapshotTest(<TextareaInfo />, 'about info');
  snapshotTest(<TextareaInfoFunction />, 'about info by function');
  snapshotTest(<TextareaFooter />, 'about footer');
  test('should render textarea element', () => {
    const { container } = render(<TextareaBase />);
    classTest(container.querySelector(textareaClassName)!, textareaFocusClassName);
    classLengthTest(container, 'textarea', 1);
  });
  test('should render default value', () => {
    const value = 'test';
    const { container } = render(<Textarea rows={6} defaultValue={value} />);
    const textarea = container.querySelector('textarea')!;
    textContentTest(textarea, value);
    attributesTest(textarea, 'rows', '6');
  });
  test('should render value', () => {
    const value = 'test';
    const { container } = render(<Textarea rows={6} value={value} />);
    const textarea = container.querySelector('textarea')!;
    textContentTest(textarea, value);
    attributesTest(textarea, 'rows', '6');
  });
  test('should render value when set value and defaultValue', () => {
    const value = 'test';
    const defaultValue = 'demo';
    const { container } = render(<Textarea defaultValue={defaultValue} value={value} />);
    const textarea = container.querySelector('textarea')!;
    textContentTest(textarea, value);
  });
  test('should render when set width', () => {
    const value = 'test';
    const { container } = render(<Textarea value={value} width={300} />);
    styleTest(container.querySelector(textareaClassName)!, 'width: 300px;');
  });
  test('should call onChange', () => {
    jest.useFakeTimers();
    const changeFn = jest.fn();
    const { container } = render(<Textarea onChange={changeFn} />);
    fireEvent.change(container.querySelector('textarea') as HTMLTextAreaElement, {
      target: { value: 'test' },
    });
    jest.runAllTimers();
    expect(changeFn.mock.calls[0][0]).toBe('test');
    expect(changeFn).toBeCalled();
  });
  test('should render when set style', () => {
    const style: React.CSSProperties = { width: 120, marginInlineEnd: 12 };
    const { container } = render(<Textarea size='small' style={style} placeholder='small size' />);
    styleTest(
      container.querySelector(textareaClassName)!,
      'width: 120px; margin-inline-end: 12px;',
    );
  });
});
describe('Textarea[Size]', () => {
  test('should render small size', () => {
    const { container } = render(<Textarea size='small' placeholder='small size' />);
    classTest(container.querySelector(textareaClassName)!, textareaSmallClassName);
  });
});
describe('Textarea[Autosize]', () => {
  test('should render double textarea to autosize', () => {
    const { container } = render(<TextareaAutosize />);
    const textareas = container.querySelectorAll('textarea');
    expect(textareas.length).toBe(2);
    classTest(textareas[1], textareaShadowClassName);
    styleTest(textareas[0], 'overflow: auto; max-height: 150px; height: 0px;');
    fireEvent.change(textareas[0] as HTMLTextAreaElement, {
      target: { value: 'test' },
    });
    textareas.forEach((textarea) => {
      textContentTest(textarea, 'test');
    });
  });
  test('should render with maxHeight', () => {
    const maxHeight = 200;
    const { container } = render(
      <Textarea autosize placeholder='input something' maxHeight={maxHeight} />,
    );
    expect(
      container
        .querySelectorAll('textarea')[0]
        .getAttribute('style')
        ?.split(';')
        .map((e) => e.trim())
        .includes(`max-height: ${maxHeight}px`),
    ).toBeTruthy();
  });
});
describe('Textarea[Info:function]', () => {
  beforeAll(() => {
    jest.useFakeTimers();
  });
  afterAll(() => {
    jest.runAllTimers();
  });
  test('should render tip', () => {
    const { container } = render(<TextareaInfoFunction />);
    fireEvent.change(container.querySelector('textarea') as HTMLTextAreaElement, {
      target: { value: 'test' },
    });
    classLengthTest(container, textareaInfoClassName, 1);
  });
  test('should render tip we want', () => {
    const { container } = render(<TextareaInfoFunction />);
    fireEvent.change(container.querySelector('textarea') as HTMLTextAreaElement, {
      target: { value: 'test' },
    });
    textContentTest(container.querySelector(textareaInfoClassName)!, 'total is  4');
    fireEvent.change(container.querySelector('textarea') as HTMLTextAreaElement, {
      target: { value: 'testtesttesttesttesttest' },
    });
    classTest(container.querySelector(textareaInfoClassName)!, textareaInfoErrorClassName);
  });
});
describe('Textarea[Info:number]', () => {
  beforeAll(() => {
    jest.useFakeTimers();
  });
  afterAll(() => {
    jest.runAllTimers();
  });
  test('should render when not set value', () => {
    const { container } = render(<Textarea info={20} />);
    expect(
      container.querySelectorAll(textareaInfoClassName).length ||
        container.querySelectorAll('.' + textareaInfoErrorClassName).length,
    ).toBe(0);
  });
  test('should render tip', () => {
    const { container } = render(<Textarea info={20} />);
    fireEvent.change(container.querySelector('textarea') as HTMLTextAreaElement, {
      target: { value: 'test' },
    });
    textContentTest(container.querySelector(textareaInfoClassName)!, '4 / 20');
  });
  test('should render error', () => {
    const { container } = render(<Textarea info={20} />);
    fireEvent.change(container.querySelector('textarea') as HTMLTextAreaElement, {
      target: { value: '1234456789012345678901' },
    });
    textContentTest(container.querySelector('.' + textareaInfoErrorClassName)!, '22 / 20');
  });
});
describe('Textarea[Trim]', () => {
  test('Textarea trim result', () => {
    jest.useFakeTimers();
    const changeFn = jest.fn();
    const { container } = render(<Textarea onChange={changeFn} trim />);
    fireEvent.blur(container.querySelector('textarea') as HTMLTextAreaElement, {
      target: { value: '   test    ' },
    });
    jest.runAllTimers();
    expect(changeFn.mock.calls[0][0].length).toBe(4);
  });
});
describe('Textarea[Disabled]', () => {
  test('Textarea disbled result', () => {
    const handleFn = jest.fn();
    const { container } = render(<Textarea defaultValue='default' disabled onChange={handleFn} />);
    expect(container.querySelector('textarea')?.getAttribute('disabled')).toBe('');
    fireEvent.focus(container.querySelector('textarea') as HTMLTextAreaElement, {
      target: { value: 'test' },
    });
    textContentTest(container.querySelector('textarea')!, 'default');
    expect(handleFn.mock.calls.length).toBe(0);
  });
});
describe('Textarea[Form Disabled]', () => {
  test('should use form disabled status first', () => {
    const handleFn = jest.fn();
    const { container } = render(
      <Form disabled>
        <Textarea disabled={false} onChange={handleFn} />
      </Form>,
    );
    hasAttributesTest(container.querySelector('form')!, 'disabled');
    fireEvent.focus(container.querySelector('textarea') as HTMLTextAreaElement, {
      target: { value: 'test' },
    });
    expect(handleFn.mock.calls.length).toBe(0);
  });
});
describe('Textarea[onBlur onChange on delay]', () => {
  test('should trigger events', () => {
    const changeHandler = jest.fn();
    const blurHandler = jest.fn();
    const { container } = render(
      <Textarea defaultValue='default' onChange={changeHandler} onBlur={blurHandler} />,
    );
    fireEvent.change(container.querySelector('textarea') as HTMLTextAreaElement, {
      target: { value: 'aaa' },
    });
    expect(changeHandler.mock.calls.length).toBe(1);
    expect(changeHandler.mock.calls[0][0]).toBe('aaa');
    fireEvent.blur(container.querySelector('textarea') as HTMLTextAreaElement, {
      target: { value: 'aaa' },
    });
    expect(blurHandler.mock.calls.length).toBe(1);
  });
});
describe('Textarea[onEnterPress]', () => {
  test('should trigger events', () => {
    const onEnterPress = jest.fn();
    const { container } = render(
      <Textarea defaultValue='default' onEnterPress={onEnterPress} onBlur={onEnterPress} />,
    );
    fireEvent.keyUp(container.querySelector('textarea') as HTMLTextAreaElement, {
      target: { value: 'aaa' },
      keyCode: 13,
    });
    expect(onEnterPress.mock.calls.length).toBe(1);
    expect(onEnterPress.mock.calls[0][0]).toBe('aaa');
  });
});
describe('Textarea[placeholderm, rows, className, style, footer, underline]', () => {
  test('should render placeholder and rows', () => {
    const { container } = render(<Textarea placeholder='input something' rows={6} />);
    const textarea = container.querySelector('textarea')!;
    attributesTest(textarea, 'rows', '6');
    attributesTest(textarea, 'placeholder', 'input something');
  });
  test('should render className and style', () => {
    const style = { color: 'blue' };
    const className = 'class-name-test';
    const { container } = render(<Textarea style={style} className={className} />);
    const textarea = container.querySelector(textareaClassName)!;
    expect(textarea).toBeTruthy();
    classTest(textarea, className);
    styleTest(textarea, 'color: blue;');
  });
  test('should renderFooter', () => {
    const { container } = render(
      <Textarea placeholder='input something' renderFooter={() => <span>hello world</span>} />,
    );
    textContentTest(container.querySelector(textareaFooterClassName)!, 'hello world');
  });
});
describe('Textarea[underline]', () => {
  test('should render underline', () => {
    const { container } = render(<Textarea placeholder='input something' underline />);
    classTest(container.querySelector(textareaClassName)!, textareaUnderlineClassName);
  });
});
describe('Textarea[BeforeChange]', () => {
  test('should render when set beforeChange', () => {
    const hanldFn = jest.fn((e) => e + 1);
    const { container } = render(<Textarea placeholder='input something' beforeChange={hanldFn} />);
    fireEvent.change(container.querySelector('textarea') as HTMLTextAreaElement, {
      target: { value: '1' },
    });
    textContentTest(container.querySelector('textarea')!, '11');
    expect(hanldFn.mock.calls.length).toBe(1);
  });
});
describe('Textarea[Border]', () => {
  test('should render when set border is false', () => {
    const { container } = render(<Textarea placeholder='input something' border={false} />);
    classTest(container.querySelector(textareaClassName)!, textareaNoBorderClassName);
  });
});
describe('Textarea[Popover]', () => {
  test('should render when set popover', () => {
    const { container } = render(
      <Textarea autosize placeholder='input something' maxHeight={150} popover tip='121' />,
    );
    fireEvent.focus(container.querySelector('textarea') as HTMLTextAreaElement, {
      target: { value: 'test' },
    });
    expect(screen.getByText('121')).toBeTruthy();
  });
});
