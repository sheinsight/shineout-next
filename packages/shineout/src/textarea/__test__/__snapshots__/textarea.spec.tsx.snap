// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Textarea[Base] should render correctly  1`] = `
<div
  class="textarea-wrapper-0-2-1 textarea-wrapperFocus-0-2-5"
  data-soui-type="input"
>
  <textarea
    class="textarea-paddingBox-0-2-2 textarea-textarea-0-2-11"
    placeholder="input something"
    rows="4"
  />
</div>
`;

exports[`Textarea[Base] should render correctly about autosize 1`] = `
<div
  class="textarea-wrapper-0-2-1"
  data-soui-type="input"
>
  <textarea
    class="textarea-paddingBox-0-2-2 textarea-textarea-0-2-11"
    placeholder="Please enter content, the height can be automatically adjusted"
    rows="2"
    style="overflow: auto; max-height: 150px; height: 0px;"
  />
  <textarea
    class="textarea-paddingBox-0-2-2 textarea-textarea-0-2-11 textarea-shadow-0-2-15"
    placeholder="Please enter content, the height can be automatically adjusted"
    rows="2"
  />
</div>
`;

exports[`Textarea[Base] should render correctly about footer 1`] = `
<div
  class="textarea-wrapper-0-2-1"
  data-soui-type="input"
>
  <textarea
    class="textarea-paddingBox-0-2-2 textarea-textarea-0-2-11"
    placeholder="input something"
    rows="3"
  />
  <div
    class="textarea-paddingBox-0-2-2 textarea-footer-0-2-16"
  >
    <div
      style="text-align: right;"
    >
      <span>
        0
      </span>
       / 20
    </div>
  </div>
</div>
`;

exports[`Textarea[Base] should render correctly about info 1`] = `
<div
  class="textarea-wrapper-0-2-1"
  data-soui-type="input"
>
  <textarea
    class="textarea-paddingBox-0-2-2 textarea-textarea-0-2-11"
    placeholder="input something"
    rows="4"
  />
</div>
`;

exports[`Textarea[Base] should render correctly about info by function 1`] = `
<div
  class="textarea-wrapper-0-2-1"
  data-soui-type="input"
>
  <textarea
    class="textarea-paddingBox-0-2-2 textarea-textarea-0-2-11"
    placeholder="input something"
    rows="4"
  />
</div>
`;

exports[`Textarea[Base] should render correctly about size 1`] = `
<div>
  <div
    class="textarea-wrapper-0-2-1 textarea-wrapperSmall-0-2-3"
    data-soui-type="input"
    style="width: 120px; margin-inline-end: 12px;"
  >
    <textarea
      class="textarea-paddingBox-0-2-2 textarea-textarea-0-2-11"
      placeholder="small size"
      rows="3"
    />
  </div>
  <div
    class="textarea-wrapper-0-2-1"
    data-soui-type="input"
    style="width: 120px; margin-inline-end: 12px;"
  >
    <textarea
      class="textarea-paddingBox-0-2-2 textarea-textarea-0-2-11"
      placeholder="default size"
      rows="3"
    />
  </div>
  <div
    class="textarea-wrapper-0-2-1 textarea-wrapperLarge-0-2-4"
    data-soui-type="input"
    style="width: 120px; margin-inline-end: 12px;"
  >
    <textarea
      class="textarea-paddingBox-0-2-2 textarea-textarea-0-2-11"
      placeholder="large size"
      rows="3"
    />
  </div>
</div>
`;
